torch/csrc/jit/codegen/cuda/kernel_db/test/test_nvfuser_kernel_db_open.cpp

```diff
@@ -0,0 +1,124 @@
+#if defined(USE_CUDA)
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <torch/torch.h>
+
+#include <torch/csrc/jit/codegen/cuda/kernel_db/kernel_db.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_db/utils.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_gpu_validator.h>
+#include <torch/csrc/jit/codegen/cuda/test/test_utils.h>
+
+// Tests go in torch::jit
+namespace torch {
+namespace jit {
+
+namespace Nvf = torch::jit::fuser::cuda;
+
+// RUN CMD: bin/test_jit --gtest_filter="NVFuserTest*KernelDb_Open*"
+TEST_F(NVFuserTest, KernelDb_Open_CUDA) {
+  // Try a DB path that should fail
+  try {
+    auto& kernel_db =
+        Nvf::KernelDb::get("../nvfuser_kernel_db_open_test", "db.csv");
+    ASSERT_FALSE(kernel_db.enabled());
+    SUCCEED();
+  } catch (const std::exception& e) {
+    FAIL() << "Unexpected failure while opening an improper Kernel DB Dir!";
+  }
+
+  // Check a corrupted DB and reset the DB
+  // 1.) Test writes a bad db.csv file and open fails to match header
+  // 2.) Should delete cubins because of bad db.csv file.
+  // 3.) Creates a new empty db.csv file with proper header
+  try {
+    const std::string kernel_db_dir("nvfuser_kernel_db_open_test");
+    const std::string bad_text("blahblahblah\n");
+    const std::string header(
+        "kernel_signature,compile_args,kernel_code_file,cubin_file");
+    fs::path test_db_path = fs::temp_directory_path() / kernel_db_dir;
+    if (fs::is_directory(test_db_path)) {
+      fs::remove_all(test_db_path);
+    }
+    ASSERT_TRUE(fs::create_directory(test_db_path));
+
+    // Setup 1
+    const std::string kernel_db_file("db.csv");
+    fs::path test_db_file = test_db_path / kernel_db_file;
+    ASSERT_FALSE(fs::is_regular_file(test_db_file));
+    ASSERT_TRUE(Nvf::copy_to_text_file(test_db_file.string(), bad_text));
+    ASSERT_TRUE(fs::is_regular_file(test_db_file));
+    // Setup 2
+    fs::path test_cubin_file = test_db_path / "test1.cubin";
+    ASSERT_TRUE(Nvf::copy_to_text_file(test_cubin_file.string(), bad_text));
+    // Execute 1, 2, 3
+    auto& kernl_db =
+        Nvf::KernelDb::get(kernel_db_dir, kernel_db_file, true, false, true);
+    // Check 1
+    ASSERT_TRUE(fs::is_regular_file(test_db_file));
+    // Check 2
+    ASSERT_FALSE(fs::is_regular_file(test_cubin_file));
+    // Check 3
+    std::ifstream db_file(test_db_file.c_str(), std::ios::in);
+    std::string line;
+    std::getline(db_file, line);
+    ASSERT_TRUE(header == line);
+
+    // Cleanup DB Directory
+    if (fs::is_directory(test_db_path)) {
+      fs::remove_all(test_db_path);
+    }
+    SUCCEED();
+  } catch (const std::exception& e) {
+    FAIL() << "Failed replacing a bad db.csv file and removing cubins!"
+           << e.what();
+  }
+
+  // Check a successful opening of an existing DB
+  try {
+    // Setup DB Directory
+    const std::string kernel_db_dir("nvfuser_kernel_db_test");
```

Comment:
This feels messy... Like we are creating so many different folders here :laughing: 
But I see that we are nuking the folder for each tests, so it's hard to share a folder for each case. :cry: